src/Snowflake/Command.php

```diff
@@ -0,0 +1,166 @@
+<?php
+
+declare(strict_types=1);
+
+namespace ProjectMigrationTool\Snowflake;
+
+use Keboola\SnowflakeDbAdapter\Connection;
+use Keboola\SnowflakeDbAdapter\QueryBuilder;
+
+class Command
+{
+
+    private const MIGRATION_SHARE_PREFIX = 'MIGRATION_SHARE_';
+
+    private const SKIP_CLONE_SCHEMAS = [
+        'INFORMATION_SCHEMA',
+        'PUBLIC',
+    ];
+
+    public static function useRole(Connection $connection, string $role): void
+    {
+        $connection->query(sprintf('USE ROLE "%s";', $role));
+    }
+
+    public static function getRegion(Connection $connection): string
+    {
+        $region = $connection->fetchAll('SELECT CURRENT_REGION() AS "region";');
+
+        return $region[0]['region'];
+    }
+
+    public static function getAccount(Connection $connection): string
+    {
+        $account = $connection->fetchAll('SELECT CURRENT_ACCOUNT() AS "account";');
+
+        return $account[0]['account'];
+    }
+
+    public static function createShare(Connection $connection, array $databases, string $destinationAccount): void
+    {
+        foreach ($databases as $database) {
+            $shareName = sprintf('%s%s', self::MIGRATION_SHARE_PREFIX, strtoupper($database));
+
+            $connection->query(sprintf(
+                'DROP SHARE IF EXISTS %s;',
+                QueryBuilder::quoteIdentifier($shareName)
+            ));
+
+            $connection->query(sprintf(
+                'CREATE SHARE %s;',
+                QueryBuilder::quoteIdentifier($shareName)
+            ));
+
+            $connection->query(sprintf(
+                'GRANT USAGE ON DATABASE %s TO SHARE %s;',
+                QueryBuilder::quoteIdentifier($database),
+                QueryBuilder::quoteIdentifier($shareName)
+            ));
+
+            $connection->query(sprintf(
+                'GRANT USAGE ON ALL SCHEMAS IN DATABASE %s TO SHARE %s;',
+                QueryBuilder::quoteIdentifier($database),
+                QueryBuilder::quoteIdentifier($shareName)
+            ));
+
+            $connection->query(sprintf(
+                'GRANT SELECT ON ALL TABLES IN DATABASE %s TO SHARE %s;',
+                QueryBuilder::quoteIdentifier($database),
+                QueryBuilder::quoteIdentifier($shareName)
+            ));
```

Comment:
Tady by měl být future grant, jinak mezi tím, co tohle vytvoříš a spustíš migraci můžou vznikat tabulky, které ten share neuvidí.

Reply:
dám ho tam... díky

Reply:
future grant bude na share problém - `Future grant on objects of type TABLE to SHARE is restricted.
`

Reply:
No to je pitomý :( 

Ale dá se holt zavolat ten GRANT SELECT ON ALL TABLES pak znova a znova a znova holt. A naposled se zavolá, až bude projekt pauznutý...